### PR TITLE
Expand beta process to perform all actions - compile, publish

### PR DESCRIPTION
**Pull request by adcreare**
Closes https://github.com/checkdigit/github-actions/issues/12
Implements:
* All in one command instead of stages
* drop the use of publish and rely on tsc only
* copy all non-typescript files to dist
* Get source maps working
* require the files property in package.json

Example usage over in ping PR https://github.com/checkdigit/ping/pull/29

### CODE REVIEW

**Review by carlansley — CHANGES_REQUESTED** [review]
nice, like it.  just one minor comment.

File: src/publish-beta/command.ts [hunk 1]
```diff
@@ -1,28 +1,22 @@
...
+import { packageJSONUpdate } from './package';
+import copyNonTSFiles from './copy-files';
+import compile from './compile';
+import publish from './publish';
 
 const log = debug('action:command');
 export default async function (): Promise<void | boolean> {
-  const command = getInput('command').toLowerCase();
-  log('Action starts:', command);
+  log('Action starts:command');
```
[the remark this answers is not in the document]
Comment (adcreare) [reply src/publish-beta/command.ts]: Yup - agreed

**Comment by adcreare** [comment]
Bunch of fixes
- Removed command file completely
- src folder is now included in package (I also parse the package.json files property and add /src/ there if it doesn't exist
- src folder is cleaned so only .ts files and swagger.yml files exist (exclusing everything else including .test.ts and .spec.ts files)

Ready for another look.

File: src/publish-beta/compile.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// publish-beta/compile.ts
+
+import childProcess from 'node:child_process';
+import util from 'node:util';
+import debug from 'debug';
+
+const log = debug('action:compile');
```
Comment (carlansley) [inline src/publish-beta/compile.ts:7]: I think all the debug namespaces should be prefixed by the name of the action, e.g. `publish-beta:compile`

**Review by carlansley — CHANGES_REQUESTED** [review]
lgtm, one minor issue re the logging

**Comment by adcreare** [comment]
Yup! Done - ready for another look

**Review by carlansley — APPROVED** [review]

**Pull request merged**